src/crimson/os/seastore/cache.h

```diff
@@ -631,24 +712,27 @@ class Cache {
   /**
    * alloc_new_extent
    *
-   * Allocates a fresh extent. if delayed is true, addr will be alloc'd later
+   * Allocates a fresh extent. if delayed is true, addr will be alloc'd later.
+   * Note that epaddr can only be fed by the btree lba unittest for now
    */
   template <typename T>
   TCachedExtentRef<T> alloc_new_extent(
     Transaction &t,         ///< [in, out] current transaction
     extent_len_t length,    ///< [in] length
     placement_hint_t hint,  ///< [in] user hint
-    rewrite_gen_t gen       ///< [in] rewrite generation
+    rewrite_gen_t gen,      ///< [in] rewrite generation
+    std::optional<paddr_t> epaddr = std::nullopt ///< [in] paddr fed by callers
```

Comment:
If its intent is (will be) only unit tests, how about using the explicit macro `#ifdef UNIT_TESTS_BUILT` ?